cloud/docker/docker.py

```diff
@@ -1110,6 +1127,15 @@ def get_differing_containers(self):
                 self.reload_reasons.append('volumes_from ({0} => {1})'.format(actual_volumes_from, expected_volumes_from))
                 differing.append(container)
 
+            # LOG_DRIVER
+
+            expected_log_driver = set(self.module.params.get('log_driver') or [])
+            actual_log_driver = set(container['HostConfig']['LogConfig']['Type'] or [])
+            if actual_log_driver != expected_log_driver:
```

Comment:
Two more things that I found:

1. The `log_driver` values here are strings, not lists of strings, so the `set()` constructor should be unnecessary (although it would technically still work, as long as no two options are anagrams :wink:)
2. This appears to trigger a false positive in the reloading logic if no log_driver was set when the container was launched:

```
"msg": "removed 1 container, stopped 1 container, started 1 container, created 1 container.", 
"reload_reasons": "log_driver (set([u'e', u'f', u'i', u'j', u'-', u'l', u'o', u'n', u's']) => set([]))", 
```

Finally -- is it possible for a dict to be missing on the path `['HostConfig']['LogConfig']['Type']`, if you're running with a older docker daemon that doesn't support log_driver? I don't have an older docker handy to test against -- I wish I had a build matrix of docker and docker-py versions available.